Plugins/Benchmark/BenchmarkPlugin.swift

```diff
@@ -128,98 +178,110 @@ import PackagePlugin
-                }
-            case "export":
-                args.append(contentsOf: ["--command", "export"])
-                if positionalArguments.count > 0 {
-                    args.append(contentsOf: ["--export-format", positionalArguments[0]])
-                }
-            case "baseline":
-                args.append(contentsOf: ["--command", "baseline"])
-                if positionalArguments.count > 0 {
-                    args.append(contentsOf: ["--baseline-name", positionalArguments[0]])
+        benchmarks.forEach { benchmark in
+            args.append(contentsOf: ["--benchmark-executable-paths", benchmark.path.string])
+        }
+
+        try withCStrings(args) { cArgs in
+            let newPath = benchmarkTool.path
+            // This doesn't work for external dependents
+            // https://forums.swift.org/t/swiftpm-always-rebuilds-command-plugins-in-release-configuration/63225
+//            let toolname = benchmarkTool.path.lastComponent
+//            let newPath = benchmarkTool.path.removingLastComponent().removingLastComponent()
+//                .appending(subpath: "release").appending(subpath: toolname)
+
+            if debug > 0 {
+                print("To debug, start BenchmarkTool in LLDB using:")
+                print("lldb \(newPath.string)")
+                print("")
+                print("Then launch BenchmarkTool with:")
+                print("run ", terminator: "")
+                for arg in 1 ..< args.count {
+                    print("\(args[arg]) ", terminator: "")
                 }
-            default:
-                print("Unknown command/option \(commandToPerform)")
+                print("")
                 return
             }
 
-            withCStrings(args) { cArgs in
-                var status = posix_spawn(&pid, benchmarkTool.path.string, nil, nil, cArgs, environ)
-
-                if status == 0 {
-                    if waitpid(pid, &status, 0) != -1 {
-                        if status != 0 {
-                            benchmarkFailure = true
-                        }
-                    } else {
-                        print("BenchmarkTool returned a non-zero exit code, errno = \(errno)")
-                        exit(errno)
+            var pid: pid_t = 0
+            var status = posix_spawn(&pid, newPath.string, nil, nil, cArgs, environ)
+
+            if status == 0 {
+                if waitpid(pid, &status, 0) != -1 {
+                    if status != 0 {
+                        print("One or more benchmark suites had a threshold violation or crashed during runtime.")
+                        throw MyError.benchmarkDeviationOrBenchmarkFailed
                     }
                 } else {
-                    print("Failed to run BenchmarkTool, posix_spawn() returned [\(status)]")
+                    print("BenchmarkTool returned a non-zero exit code, errno = \(errno)")
```

Comment:
But that is the status returned from posix_spawn, not waitpid ?

Reply:
If I don't mess with indentation we enter this block when `waitpid()` returned `-1`

The man page reads
```
If wait3(), wait4(), or waitpid() returns due to a stopped or terminated child process, the process ID of the child is returned to the calling process.  If there are no children not previously awaited, -1 is returned with
     errno set to [ECHILD].  Otherwise, if WNOHANG is specified and there are no stopped or exited children, 0 is returned.  If an error is detected or a caught signal aborts the call, a value of -1 is returned and errno is set
     to indicate the error.
```